src/lib/services/account/lcd.ts

```diff
@@ -18,3 +21,19 @@ export const getAccountDataLcd = async (
     icns,
   };
 };
+
+export const getAccountTypeLcd = async (
+  endpoint: string,
+  address: BechAddr
+) => {
+  const accountTypeLcd = await axios
+    .get(`${endpoint}/cosmos/auth/v1beta1/accounts/${address}`)
+    .then((res) => parseWithError(zAccountTypeLcd, res.data));
+
+  const isContract = await axios
+    .get(`${endpoint}/cosmwasm/wasm/v1/contract/${address}`)
+    .then(() => true)
+    .catch(() => false);
```

Comment:
You can wrap both `accountTypeLcd` and `isContract` with `Promise.all` since both are not related to each others.